core/src/Kokkos_Array.hpp

```diff
@@ -158,6 +159,72 @@ struct Array {
     return &m_internal_implementation_private_member_data[0];
   }
 
+  template <size_t I>
+  KOKKOS_INLINE_FUNCTION
+      typename std::enable_if<(I <= N),
+                              typename std::add_lvalue_reference<T>::type>::type
+      get() & {
+    return m_internal_implementation_private_member_data[I];
+  }
+
```

Comment:
Same as with pair: introduction of a member function get where the `std::array` doesn't have that. Should be resolved the same way as for `Kokkos::pair::get`